### ISSUE TITLE
How to fix certain dimensions of lengthscale in optimization 

### ISSUE THREAD

**Issue by HiroIshida** [cornellius-gp/gpytorch#1169]
Is there any way (or hints) to fix certain dimensions of lengthscale in optimization? For example, say I have a kernel with lengthscale of `[0.1, 0.3, 0.3]`, and what I want to do is to optimize only the first dimension of that and the other dimensions be static. 

A naive workaround would be using both `GreaterThan` and `LessThan` constraints, but this is not equiavalent to keeping some parameter fixed, because those constraints just do clip the value after the optimization according to https://github.com/cornellius-gp/gpytorch/issues/849.

**Comment by KeAWang** [comment]
I don't think there's a way right now to do this since kernel parameters are usually stored as tensors. You could however implement your own version that separates each dimension of a  parameter into its own tensor.

Personally I would just reset the fixed parameter to the desired value after each gradient step, which require less effort.

**Comment by gpleiss** [comment]
Yeah I think @KeAWang 's suggestion is the cleanest option. After `optimizer.step()`, you can manually update the lengthscale dimensions that you care about.

I think something like this should work:

```python
optimizer.step()
lengthscale = model.covar_module.base_kernel.lengthscale
lengthscale[..., 2] = 0.3  # or whatever value you care about
model.covar_module.base_kernel.lengthscale = lengthscale
```

@HiroIshida let us know if this works for you!

**Comment by HiroIshida** [comment]
@gpleiss 
Thank you very much, the method you suggested worked perfectly well
